chia/data_layer/s3_plugin_service.py

```diff
@@ -47,26 +62,27 @@ def __init__(
             aws_secret_access_key=aws_secret_access_key,
         )
         self.store_ids = store_ids
-        self.buckets = buckets
-        self.urls = urls
         self.instance_name = instance_name
 
-    async def check_store_id(self, request: web.Request) -> web.Response:
+    async def handle_upload(self, request: web.Request) -> web.Response:
         self.update_instance_from_config()
         try:
             data = await request.json()
         except Exception as e:
             print(f"failed parsing request {request} {e}")
-            return web.json_response({"handles_url": False})
-        store_id = bytes32.from_hexstr(data["id"])
-        if store_id in self.store_ids:
-            return web.json_response({"handles_store": True})
-        return web.json_response({"handles_store": False})
+            return web.json_response({"handle_upload": False})
+
+        id = bytes32.from_hexstr(data["store_id"])
+        for store_id in self.store_ids:
+            if store_id.id == id and store_id.bucket:
+                return web.json_response({"handle_upload": True, "bucket": store_id.bucket})
```

Comment:
```suggestion
        store_id = bytes32.from_hexstr(data["store_id"])
        for store in self.store_ids:
            if store.id == store_id and store.bucket:
                return web.json_response({"handle_upload": True, "bucket": store_id.bucket})
```

Reply:
if you take my suggestion on using stores instead of store_ids this suggestion will break

Reply:
changed correctly I beleive